Mathlib/Analysis/NormedSpace/PiTensorProduct/InjectiveSeminorm.lean

```diff
@@ -0,0 +1,460 @@
+  rw [injectiveSeminorm]
+  refine csSup_le ?_ ?_
+  · existsi 0
+    simp only [Set.mem_setOf_eq]
+    existsi PUnit, inferInstance, inferInstance
+    ext x
+    simp only [Seminorm.zero_apply, Seminorm.comp_apply, coe_normSeminorm]
+    have heq : toDualContinuousMultilinearMap (F := PUnit) x = 0 := by ext _
+    rw [heq, norm_zero]
+  · intro p hp
+    simp only [Set.mem_setOf_eq] at hp
+    obtain ⟨G, _, _, h⟩ := hp
+    rw [h]; intro x; simp only [Seminorm.comp_apply, coe_normSeminorm]
+    exact toDualContinuousMultilinearMap_le_projectiveSeminorm _
+
+theorem injectiveSeminorm_tprod_le (m : Π (i : ι), E i) :
+    injectiveSeminorm (⨂ₜ[𝕜] i, m i) ≤ ∏ i, ‖m i‖ :=
+  le_trans (injectiveSeminorm_le_projectiveSeminorm _) (projectiveSeminorm_tprod_le m)
+
+noncomputable instance : SeminormedAddCommGroup (⨂[𝕜] i, E i) :=
+  AddGroupSeminorm.toSeminormedAddCommGroup injectiveSeminorm.toAddGroupSeminorm
+
+noncomputable instance : NormedSpace 𝕜 (⨂[𝕜] i, E i) where
+  norm_smul_le a x := by
+    change injectiveSeminorm.toFun (a • x) ≤ _
+    rw [injectiveSeminorm.smul']
+    rfl
+
+variable (𝕜 E F)
+
+/-- The linear equivalence between `ContinuousMultilinearMap 𝕜 E F` and `(⨂[𝕜] i, Eᵢ) →L[𝕜] F`
+induced by `PiTensorProduct.lift`, for every normed space `F`.
+-/
+@[simps]
+noncomputable def liftEquiv : ContinuousMultilinearMap 𝕜 E F ≃ₗ[𝕜] (⨂[𝕜] i, E i) →L[𝕜] F where
+  toFun f := LinearMap.mkContinuous (lift f.toMultilinearMap) ‖f‖
+    (fun x ↦ norm_eval_le_injectiveSeminorm f x)
+  map_add' f g := by ext _; simp only [ContinuousMultilinearMap.toMultilinearMap_add, map_add,
+    LinearMap.mkContinuous_apply, LinearMap.add_apply, ContinuousLinearMap.add_apply]
+  map_smul' a f := by ext _; simp only [ContinuousMultilinearMap.toMultilinearMap_smul, map_smul,
+    LinearMap.mkContinuous_apply, LinearMap.smul_apply, RingHom.id_apply,
+    ContinuousLinearMap.coe_smul', Pi.smul_apply]
+  invFun l := MultilinearMap.mkContinuous (lift.symm l.toLinearMap) ‖l‖ (fun x ↦ by
+    simp only [lift_symm, LinearMap.compMultilinearMap_apply, ContinuousLinearMap.coe_coe]
+    refine le_trans (ContinuousLinearMap.le_opNorm _ _) (mul_le_mul_of_nonneg_left ?_
+      (norm_nonneg l))
+    exact injectiveSeminorm_tprod_le x)
+  left_inv f := by ext x; simp only [LinearMap.mkContinuous_coe, LinearEquiv.symm_apply_apply,
+      MultilinearMap.coe_mkContinuous, ContinuousMultilinearMap.coe_coe]
+  right_inv l := by
+    rw [← ContinuousLinearMap.coe_inj]
+    apply PiTensorProduct.ext; ext m
+    simp only [lift_symm, LinearMap.mkContinuous_coe, LinearMap.compMultilinearMap_apply,
+      lift.tprod, ContinuousMultilinearMap.coe_coe, MultilinearMap.coe_mkContinuous,
+      ContinuousLinearMap.coe_coe]
+
+/-- The equivalence `PiTensorProduct.liftEquiv` as an isometric linear equivalence.
+-/
+noncomputable def liftIsometry  : ContinuousMultilinearMap 𝕜 E F ≃ₗᵢ[𝕜] (⨂[𝕜] i, E i) →L[𝕜] F :=
+  {liftEquiv 𝕜 E F with
```

Comment:
```suggestion
  { liftEquiv 𝕜 E F with
```
(Also, all following lines will miss one space of indentation). The rule is: if you define a set, no space after {. Otherwise, put a space. Same thing for the closing accolade (comment on dit en anglais? curly brace?), by the way, is needs a space before it.

Reply:
Ok. I also found a set that had a `{` with a space, so I deleted that.

As for the name of the sign, wikipedia says: `{` and `}` are "braces" in both American and British English, and also "curly brackets" in the latter.